wp-parsely.php

```diff
@@ -1649,29 +1646,19 @@ private function get_current_url( $post = 'nonpost', $post_id = 0 ) {
 			 * @param int    $post_id   ID of the post you want to get the URL for. May be 0, so $permalink will be
 			 *                          for the global $post.
 			 */
-			$permalink        = apply_filters( 'wp_parsely_permalink', $permalink, $post, $post_id );
-			$parsed_canonical = wp_parse_url( $permalink );
-			// handle issue if wp_parse_url doesn't return good host & path data, fallback to page url as a last resort.
-			if ( isset( $parsed_canonical['host'], $parsed_canonical['path'] ) ) {
-				$canonical = $scheme . $parsed_canonical['host'] . $parsed_canonical['path'];
-			} elseif ( isset( $_SERVER['HTTP_HOST'], $_SERVER['REQUEST_URI'] ) ) { // Input var okay.
-				$canonical = $scheme . sanitize_text_field( wp_unslash( $_SERVER['HTTP_HOST'] ) ) . sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ) ); // Input var okay.
-			}
+			$url = apply_filters( 'wp_parsely_permalink', $permalink, $post, $post_id );
+		} else {
+			$request_uri = isset( $_SERVER['REQUEST_URI'] )
+					? sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ) )
+					: '';
 
-			return $canonical;
+			$url = home_url( $request_uri );
 		}
-		$page_url = site_url( null, $scheme );
```

Comment:
Previously, this uses `site_url`. However, since we're dealing with the canonical links (front-end links), not WordPress file links, the correct usage should be `home_url`.